sdk/framework/openapi.go

```diff
@@ -321,9 +320,24 @@ func documentPath(p *Path, specialPaths *logical.Paths, requestResponsePrefix st
 			return strings.ToLower(pi.Parameters[i].Name) < strings.ToLower(pi.Parameters[j].Name)
 		})
 
+		uniqueOperationIdentifiers := make(map[string]struct{})
+
 		// Process each supported operation by building up an Operation object
 		// with descriptions, properties and examples from the framework.Path data.
 		for opType, opHandler := range operations {
+			operationID := constructRequestResponseIdentifier(opType, mountPathWithPrefix, path)
+
+			// Certain paths contain duplicate operation ID's for identical operations.
+			// For example, the following all map to the same operation:
+			//      PUT /v1/gcpkms/keys/deregister/:key
+			//     POST /v1/gcpkms/keys/deregister/:key
+			//   DELETE /v1/gcpkms/keys/deregister/:key
+			if _, ok := uniqueOperationIdentifiers[operationID]; ok {
+				continue
+			} else {
+				uniqueOperationIdentifiers[operationID] = struct{}{}
+			}
```

Comment:
This code is very problematic. It means that some operations will be omitted from the OpenAPI entirely. That's not good at all, when people may want to use it to underand which methods an endpoint actually supports in full.

Furthermore, it looks to me like which operations get excluded is random as it is dependent on map iteration order!

Reply:
Yes, I agree, this is not a great solution, I'll try to think about a better one, perhaps just create another mapping for the few exceptions where this applies.

Reply:
I replaced this with a (hopefully) better solution. There are 4 known cases where this exception needs to be applied. I added a flag for these cases in the lookup map.

Reply:
Sorry, but the new solution is still very problematic for my usecase, as I rely on the OpenAPI to determine the complete set of operations that a given path supports, and this would erase that information.

Additionally, it is likely to prove confusing to people referring to the OpenAPI or Swagger UI to guide their understanding of already-written code, and find it is calling an operation which is now erased, and so seems mysterious and undocumented.